### PR TITLE
Content Manager - Create components' helper functions 

### PR DESCRIPTION
**Pull request by tsarquis88**
|Related issue|
|---|
|#20528 |

## Description

This pull request moves some methods to common places, to reuse code between the different components.

Change log:
- Create the `componentsHelper.hpp` file with a method for pushing the components' status.
- Create the `hashFile()` method in the `hashUtils.hpp` file.
- UTs added.

### CODE REVIEW

File: src/shared_modules/content_manager/src/components/componentsHelper.hpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+     *
+     * @param componentName Name of the component.
+     * @param componentStatus Status to be pushed.
+     * @param context Reference to the component context.
+     */
+    static void pushComponentStatus(const std::string& componentName,
+                                    const ComponentStatus& componentStatus,
+                                    UpdaterContext& context)
+    {
+        // LCOV_EXCL_START
```
Comment (tsarquis88) [inline src/shared_modules/content_manager/src/components/componentsHelper.hpp:44]: Note: The LCOV exclusion is due to this strange report:
image

File: src/shared_modules/content_manager/src/components/APIDownloader.hpp [hunk 2]
```diff
@@ -13,6 +13,7 @@
 
 #include "../sharedDefs.hpp"
 #include "IURLRequest.hpp"
+#include "componentsHelper.hpp"
 #include "updaterContext.hpp"
 #include "utils/chainOfResponsability.hpp"
 #include <iostream>
```
Comment (jftuduri) [inline src/shared_modules/content_manager/src/components/APIDownloader.hpp:19]: Unused header
```suggestion
```

File: src/shared_modules/content_manager/src/components/componentsHelper.hpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+#ifndef _COMPONENTS_HELPER_HPP
+#define _COMPONENTS_HELPER_HPP
+
+#include "json.hpp"
+#include "updaterContext.hpp"
+#include <map>
+#include <stdexcept>
+#include <string>
+
+namespace Utils
```
Comment (jftuduri) [inline src/shared_modules/content_manager/src/components/componentsHelper.hpp:21]: I feel this helper function is not generic enough for this namespace. Searching for "namespace Utils" in the codebase shows some patterns base implementation, file manipulation classes, etc.
But, not sure what would be a better name. What is more related to? updater? pub-sub?
What do you think?

File: src/shared_modules/utils/tests/input_files/hashHelper/data.json
Comment (jftuduri) [inline src/shared_modules/utils/tests/input_files/hashHelper/data.json:1]: This file should be larger than the 4KB that the hashFile function is using as a buffer to be sure that the loop is working correctly.

File: src/shared_modules/utils/tests/hashHelper_test.cpp [hunk 4]
```diff
@@ -83,3 +89,25 @@ TEST_F(HashHelperTest, HashHelperHashIterativeSha256)
...
+
+/**
+ * @brief Test the hashing of a file.
+ *
+ */
+TEST_F(HashHelperTest, HashFile)
+{
+    // SHA1SUM as string: 9c8ff5a046322fd04e244ce92e991118f26403f1
+    const std::vector<unsigned char> expectedHash {156, 143, 245, 160, 70, 50, 47,  208, 78, 36,
+                                                   76,  233, 46,  153, 17, 24, 242, 100, 3,  241};
```
Comment (jftuduri) [inline src/shared_modules/utils/tests/hashHelper_test.cpp:101]: We could use hex literals here to make it easier to compare with the hash of the comment above:
```suggestion
const std::vector<unsigned char> expectedHash {0x9c, 0x8f,...
```

Comment (tsarquis88) [reply src/shared_modules/utils/tests/hashHelper_test.cpp]: Done!

File: src/shared_modules/utils/tests/input_files/hashHelper/data.json
Comment (tsarquis88) [reply src/shared_modules/utils/tests/input_files/hashHelper/data.json:1]: Yes. I've replaced the old file with one with a size of 5MB

File: src/shared_modules/content_manager/src/components/componentsHelper.hpp (hunk 3 above)
Comment (tsarquis88) [reply src/shared_modules/content_manager/src/components/componentsHelper.hpp]: Yes, it's true. I made some renaming.

File: src/shared_modules/content_manager/src/components/APIDownloader.hpp (hunk 2 above)
Comment (tsarquis88) [reply src/shared_modules/content_manager/src/components/APIDownloader.hpp]: Done in rebase.

File: src/shared_modules/content_manager/src/components/componentsHelper.hpp [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+#include <stdexcept>
+#include <string>
+
+namespace Components
+{
+    /**
+     * @brief Possible components' status.
+     *
+     */
+    enum Status
```
Comment (Damian-Mangold) [inline src/shared_modules/content_manager/src/components/componentsHelper.hpp:27]: Improve: I think you can use the `enum class` here instead of `enum`

**Review by jftuduri — APPROVED** [review]

File: src/shared_modules/content_manager/src/components/componentsHelper.hpp (hunk 5 above)
Comment (tsarquis88) [reply src/shared_modules/content_manager/src/components/componentsHelper.hpp]: Didn't know that. Done!

**Pull request merged**